src/vs/platform/browser/contextScopedHistoryWidget.ts

```diff
@@ -13,7 +13,8 @@ import { ContextKeyExpr, IContextKey, IContextKeyService, IContextKeyServiceTarg
 import { KeybindingsRegistry, KeybindingWeight } from 'vs/platform/keybinding/common/keybindingsRegistry';
 
 export const HistoryNavigationWidgetContext = 'historyNavigationWidget';
-export const HistoryNavigationEnablementContext = 'historyNavigationEnabled';
+const HistoryNavigationForwardsEnablementContext = 'historyNavigationForwardsEnabled';
```

Comment:
Any reason for breaking the original context into two?

Reply:
We want to allow navigating the history backwards when the cursor is at the first position, and forwards when it's at the last. This matches the behavior of SCM's implementation of history (which I looked into reusing but is very coupled to SCM)